Migrate existing mi300 runners to new mi325 capacity.

deedongala opened the pull request:
Runner capacity is shifting from MI300 to MI325 machines. These machines are compatible enough for testing, though benchmarks will need new baselines.

deedongala committed:
Update pkgci_test_amd_mi300.yml

Signed-off-by: deedongala <deekshitha.dongala@amd.com>

deedongala committed:
Update pkgci_test_sharktank.yml

Signed-off-by: deedongala <deekshitha.dongala@amd.com>

ScottTodd commented on .github/workflows/pkgci_test_amd_mi300.yml:
```diff
@@ -19,7 +19,7 @@ on:
 
 jobs:
   test_mi300:
```
This job could be renamed to mi325

```suggestion
  test_mi325:
```

ScottTodd replied:
Oh, and we can rename this entire file too, along with the references in https://github.com/iree-org/iree/blob/main/.github/workflows/pkgci.yml

saienduri replied:
Cool, @deedongala can you re-name these

deedongala replied:
I've renames all references.

ScottTodd commented on .github/workflows/pkgci_test_sharktank.yml:
```diff
@@ -53,7 +53,7 @@ jobs:
             iree_test_files: /shark-cache/data/iree-regression-cache
             sku: mi300
```
This could be changed to mi325, or we could change to "gfx target" and use gfx942 to apply to either

Fine to keep as mi300 in this PR. Looks like quite a bit of plumbing for "sku" and "mi300" over in https://github.com/iree-org/iree-test-suites, which this uses (search: https://github.com/search?q=repo%3Airee-org%2Firee-test-suites%20sku&type=code)

deedongala replied:
Yes, keeping it as mi300 in this PR makes sense, especially given the existing plumbing around sku and mi300 in iree-test-suites.

ScottTodd reviewed (commented):
Generally LGTM (especially given that mi300 runners are going offline soon, so this migration is forced). Please remember to remove "DO NOT MERGE" from the PR title once testing completes. You could also start the PR as a draft instead of changing the title like that.

deedongala committed:
Update .github/workflows/pkgci_test_amd_mi300.yml

Co-authored-by: Scott Todd <scott.todd0@gmail.com>
Signed-off-by: deedongala <deekshitha.dongala@amd.com>

deedongala committed:
Changes to be committed:
	modified:   .github/workflows/pkgci.yml
	renamed:    .github/workflows/pkgci_test_amd_mi300.yml -> .github/workflows/pkgci_test_amd_mi325.yml

Signed-off-by: deedongala <deekshitha.dongala@amd.com>
